### ISSUE TITLE
Repeating header/footer functionality

### ISSUE THREAD

**Issue by alexleonard** [jonnnnyw/php-phantomjs#99]
Creating a new issue as requested in: https://github.com/jonnnnyw/php-phantomjs/issues/52

It would be great to have a built in process for adding repeating header/footer to `createPdfRequest()`

Thanks!

**Comment by alexleonard** [comment]
Hey, just checking when this feature might be implemented? Let me know if I can do anything to help.

**Comment by jonnnnyw** [comment]
It's been implemented in master but not tagged off. I have a couple more things I want to put in the next release so hopefully will be out next week.

**Comment by jonnnnyw** [comment]
Actually it's not in master yet it's on my local. There is a breaking test. Will get onto it.

**Comment by alexleonard** [comment]
Awesome! Thanks for the reply. It will be a very well timed addition for
the project I'm working on :)

Alex Leonard ~ alex@leonard.ie ~ plus.google.com/+AlexLeonard
Websites: alex.leonard.ie ~ ebauche.net <http://www.ebauche.net>

**Comment by jonnnnyw** [comment]
I have just released v4.5 which contains this functionality. I will update the documentation tomorrow but in the meantime this is the basic usage:

```
$client = $this->getClient();

$request  = $client->getMessageFactory()->createPdfRequest();
$response = $client->getMessageFactory()->createResponse();

$request->setRepeatingHeader('<h1>Header <span style="float:right">%pageNum% / %pageTotal%</span></h1>'[, height]);
$request->setRepeatingFooter('<footer>Footer <span style="float:right">%pageNum% / %pageTotal%</span></footer>'[, height]);
```

The second parameter, height, is optional and defaults to '1cm'. The `%pageNum%` and `%pageTotal%` placeholders can be used to show total and current page numbers.

Let me know if you have any issues with it.

**Comment by alexleonard** [comment]
Hey @jonnnnyw Finally got a chance to work with this and it worked great straight out of the box! Totally fantastic.  :) :+1:
